### PR TITLE
add suboptions in ovirt docs

### LINKED ISSUE #53797: Use suboptions in oVirt modules doc
##### SUMMARY

Use suboptions in oVirt module documentations of dictionary parameters.

##### ISSUE TYPE
- Documentation Report

##### COMPONENT NAME

ovirt

##### ANSIBLE VERSION

```paste below

```

##### CONFIGURATION

```paste below

```

### PR DESCRIPTION
**Pull request by mnecas**
##### SUMMARY

Fixes #53797 

First part of suboptions.

##### ISSUE TYPE

- Docs Pull Request

##### COMPONENT NAME

ovirt
##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

**Comment by mnecas** [comment]
@machacekondra @mwperina

**Comment by mnecas** [comment]
+label ovirt

File: lib/ansible/plugins/doc_fragments/ovirt.py [hunk 1]
```diff
@@ -30,25 +30,46 @@ class ModuleDocFragment(object):
...
+                    - Token to be used instead of login with username/password.
+                    - Default value is set by I(OVIRT_TOKEN) environment variable.
+            insecure:
+                description:
+                    - A boolean flag that indicates if the server TLS certificate and host name should be checked.
+            ca_file:
+                description:
+                    - A PEM file containing the trusted CA certificates.
+                    - The certificate presented by the server will be verified using these CA certificates.
+                    - If `C(ca_file)` parameter is not set, system wide CA certificate store is used.
```
Comment (gundalow) [inline lib/ansible/plugins/doc_fragments/ovirt.py]:
```suggestion
                    - If C(ca_file) parameter is not set, system wide CA certificate store is used.
```

File: lib/ansible/plugins/doc_fragments/ovirt.py [hunk 2]
```diff
@@ -30,25 +30,46 @@ class ModuleDocFragment(object):
...
+        suboptions:
+            username:
+                description:
+                    - The name of the user, something like I(admin@internal).
+                    - Default value is set by I(OVIRT_USERNAME) environment variable.
+                required: true
+            password:
+                description:
+                    - The password of the user.
+                    - Default value is set by I(OVIRT_PASSWORD) environment variable.
```
Comment (gundalow) [inline lib/ansible/plugins/doc_fragments/ovirt.py]: We use `C(..) ` for code and environment variables. 

```suggestion
                    - Default value is set by C(OVIRT_PASSWORD) environment variable.
```

File: lib/ansible/plugins/doc_fragments/ovirt.py [hunk 3]
```diff
@@ -30,25 +30,46 @@ class ModuleDocFragment(object):
...
+                required: true
+            url:
+                description:
+                    - A string containing the API URL of the server, usually something like `I(https://server.example.com/ovirt-engine/api)`.
+                    - Default value is set by I(OVIRT_URL) environment variable.
```
Comment (gundalow) [inline lib/ansible/plugins/doc_fragments/ovirt.py]:
```suggestion
                    - Default value is set by C(OVIRT_URL) environment variable.
```

File: lib/ansible/plugins/doc_fragments/ovirt.py [hunk 4]
```diff
@@ -30,25 +30,46 @@ class ModuleDocFragment(object):
...
+                    - Either C(url) or C(hostname) is required.
+            hostname:
+                description:
+                    - A string containing the hostname of the server, usually something like `I(server.example.com)`.
+                    - Default value is set by I(OVIRT_HOSTNAME) environment variable.
+                    - Either C(url) or C(hostname) is required.
+            token:
+                description:
+                    - Token to be used instead of login with username/password.
+                    - Default value is set by I(OVIRT_TOKEN) environment variable.
```
Comment (gundalow) [inline lib/ansible/plugins/doc_fragments/ovirt.py]:
```suggestion
                    - Default value is set by C(OVIRT_TOKEN) environment variable.
```

File: lib/ansible/plugins/doc_fragments/ovirt.py [hunk 5]
```diff
@@ -30,25 +30,46 @@ class ModuleDocFragment(object):
...
+            insecure:
+                description:
+                    - A boolean flag that indicates if the server TLS certificate and host name should be checked.
+            ca_file:
+                description:
+                    - A PEM file containing the trusted CA certificates.
+                    - The certificate presented by the server will be verified using these CA certificates.
+                    - If `C(ca_file)` parameter is not set, system wide CA certificate store is used.
+                    - Default value is set by I(OVIRT_CAFILE) environment variable.
```
Comment (gundalow) [inline lib/ansible/plugins/doc_fragments/ovirt.py]:
```suggestion
                    - Default value is set by C(OVIRT_CAFILE) environment variable.
```

**Comment by gundalow** [comment]
Thanks for doing this, it makes the documentation a lot more readable.
Perhaps in this PR you could update `BOTMETA.yml` so labels and maintainers will automatically be set on this file?

File: lib/ansible/plugins/doc_fragments/ovirt.py [hunk 6]
```diff
@@ -30,25 +30,46 @@ class ModuleDocFragment(object):
...
-            certificate and host name should be checked."
-            - "C(ca_file) - A PEM file containing the trusted CA certificates. The
-            certificate presented by the server will be verified using these CA
-            certificates. If `C(ca_file)` parameter is not set, system wide
-            CA certificate store is used. Default value is set by I(OVIRT_CAFILE) environment variable."
-            - "C(kerberos) - A boolean flag indicating if Kerberos authentication
-            should be used instead of the default basic authentication."
-            - "C(headers) - Dictionary of HTTP headers to be added to each API call."
+        suboptions:
+            username:
```
Comment (gundalow) [inline lib/ansible/plugins/doc_fragments/ovirt.py:34]: Could you please add `type:` for all of the arguments. This is something we are slowly adding to all modules to improve the module docs.

**Review by gundalow — COMMENTED** [review]
Just a few minor formatting comments, I realise most of these existing in the original code though so his feels like a good point to tidy them up.

**Comment by acozine** [comment]
Thanks @mnecas for updating these!

**Pull request merged**